New doc

M0ses opened the pull request.

hennevogel commented on CONTRIBUTING.md:
```diff
@@ -1,13 +1,25 @@
+8. [Setup an OBS backend for development](#setup_backend)
+9. [Quick Start Guid (Howto setup a developer VM)](#quick_start)
+
+# <a name="request"/> Request for contributions
```
Why do you introduce those html anchors?

M0ses replied:
sure ?

https://github.com/M0ses/testing_markdown/blob/master/README.md#Test%20for%20TOC

hennevogel replied:
If you move your mouse left of the title you see the real anchor. It's lowercase and with hyphens :-)

M0ses replied:
@hennevogel  THX, will fix this

hennevogel commented on CONTRIBUTING.md:
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
 
 * Go to the next rubocop offense.
+
+# <a name="setup_backend"/>Setup an OBS backend for development
```
Vagrant includes the instructions on how to run a backend right? Why duplicate this?

adrianschroeter replied:
You may develop also in backend parts and esp. with distributed setup. So we need some place somewhere how the backend gets setup.

M0ses replied:
No, it's not the same , e.g.  start_test_backend
* does not start workers
* runs bs_publish in testmode 
* etc.

hennevogel replied:
Then you have to make clear what to do when. Otherwise people will be confused.

hennevogel commented on README.md:
```diff
@@ -27,326 +27,36 @@ The OBS source code repository is hosted on [Github](http://github.com/opensuse/
 ## Setup
 There are 3 scenarios for which you can setup an OBS instance. Running it in *production* for your users, for *development* on it and for executing the *test* suite.
```
Drop the *test suite* case, this is covered in CONTRIBUTING.md

adrianschroeter replied:
Actually, I would also drop development here, just pointing developers to the CONTRIBUTING file.
Just saying "production" is what you want.

hennevogel commented on CONTRIBUTING.md:
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
+    vagrant plugin install vagrant-exec
+    vagrant plugin install vagrant-reload
+    # optional if you are running vagrant with libvirt (e.g. kvm)
+    vagrant plugin install vagrant-libvirt
```
I would skip this as people following the rest of the README won't need it...

hennevogel commented on CONTRIBUTING.md:
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
+
+1. Install [Vagrant](https://www.vagrantup.com/downloads.html) and [VirtualBox](https://www.virtualbox.org/wiki/Downloads). Both tools support Linux, MacOS and Windows and in principal setting up your OBS development environment works similar.
+
+2. Install [vagrant-exec](https://github.com/p0deje/vagrant-exec):
```
+2. Install Vagrant plugins

hennevogel commented on CONTRIBUTING.md:
```diff
@@ -67,3 +79,74 @@ If you want to take part of this please follow this process:
+Check [src/backend/README](https://github.com/openSUSE/open-build-service/blob/master/src/backend/README) how to run the backend from the source code repository.
+
+
+# <a name="quick_start"/>Quick Start Guid (Howto setup a developer VM)
```
*Guide*

M0ses replied:
THX

hennevogel commented on CONTRIBUTING.md:
```diff
@@ -28,22 +40,22 @@ We are using priority labels from **P0** to **P4** for our issues. So if you are
 * P4: Low  - Fix this when you don't see any issues with the other priorities.
 
-# Contribute to the OBS documentation
+# <a name="contribute_docu" />Contribute to the OBS documentation
```
+How to contribute documentation

hennevogel commented on dist/README.SETUP:
```diff
@@ -1,45 +1,49 @@
 # Manual Setup Guide
```
Why do we have both? README.SETUP.md and README.SETUP

M0ses replied:
* README.SETUP.md for nice view on github
* link README.SETUP for "backwards compatiblity"

hennevogel replied:
But README.SETUP is is not a link...

hennevogel commented:
In general README.SETUP.md is exactly what I wanted to avoid:  The only content that has changed for our users is that we have introduced a 5th method of installing OBS (setup-appliance.sh)...

M0ses commented:
@hennevogel : 

* I removed redundant documentation (README.md/README.SETUP)
* I added recommendations when to use which way
* The README.md became much clearer

And no, we did not introduced the 5th way - There are 3 ways of installation with clear recommendations and example which one to use to achive a specific goal.

hennevogel commented:
@M0ses The content (not the structure or location files etc.) before was

* install via appliance
* install via packages

Now the content is 

* install via appliance
* install via setup-appliance.sh
* install via packages

hennevogel commented:
So what are we going to do about this?

hennevogel commented:
Talked to @M0ses offline. We're tracking the missing things in [trello](https://trello.com/c/ot4mIxke)
